teps/0057-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,514 @@
+
+### Versioning Resources
+
+Resources definition evolves across time. A new version of the image used is published,
+a new feature in tektoncd/pipeline is available, or a new feature / behaviour is added.
+On the other hand, the catalog should guarantee as much stability as possible for its user.
+
+A version is an identifiable information of a resource along with its kind (Task, Pipeline)
+and name, and it should be present in the definition of the resource. The location of a resource
+in the catalog should be computable given its Kind, Name and the Version.
+e.g. `/{kind}/{name}/{version}/{name}.yaml` - `pipeline/<pipeline-name>/0.5/<pipeline-name>.yaml`
+
+Adding a new version of a resource should be considered as a release of that resource. Here are some cases where a version bump for pipeline may be required
+
+- Task used in the pipeline has been deprecated/removed
+- Location of the task used in the pipeline is changed
+- If there's a new version of the task then it completely depends on the author
+  of the pipeline whether to bump the version of the pipeline or not
+- If anything in the pipeline is changed including params, results, when
+  expressions (which might change the behavior of the pipeline)
+
+#### Questions
+
+- What should be the state of pipeline if a task used in the pipeline is deprecated ?
+
+### Ownership
+
+Please address the Ownership information from [TEP(0003)- Tekton Catalog Organization](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#ownership)
+
+### Organization
+
+```bash=
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
+# The README of the resource (what is the resource about, list of versions, …)
+./{resource-type}/{resource-name}/README.md
+
+# The resource itself
+./{resource-type}/{resource-name}/{version}/{resource}.yaml
+
+# The README of the versioned resource (usage, …)
+./{resource-type}/{resource-name}/{version}/README.md
+
+# The install script which will install all the related resources
+./{resource-type}/{resource-name}/{version}/install-script.sh
+
+# Tasks directory which refers to the task used in the pipeline
+./{resource-type}/{resource-name}/{version}/tasks/..
+
+# Optional: Test working samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/tests/…
+# Optional: Addition samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/samples/…
+```
+
+For example:
+
+```bash=
+
+./pipeline/                     👈 The kind of the resource
+
+    /p1                         👈 Definition file must have same name
+       /OWNERS                  👈 owners of this resource
+       /0.1
+         /README.md
+         /install-script.sh     👈 To install all resources related to pipeline
```

Comment:
I guess if we go with bundle solution we might not need this script, because user would just have to install the pipeline and not worry about tasks anymore

Reply:
exactly!!! :D that sounds great

Reply:
bumping this, i think we can remove the install-script and tasks dir now from this example and the list above?

Reply:
Yes sure I'll update the example thanks 👍🏻